src/shared/Microsoft.Git.CredentialManager/Authentication/OAuth/OAuth2SystemWebBrowser.cs

```diff
@@ -42,55 +42,35 @@ public OAuth2SystemWebBrowser(OAuth2WebBrowserOptions options)
             _options = options;
         }
 
-        public async Task<Uri> GetAuthenticationCodeAsync(Uri authorizationUri, Uri redirectUri, CancellationToken ct)
+        public Uri UpdateRedirectUri(Uri uri)
```

Comment:
This is so much like MSAL ....

Reply:
Looks like this is the only way to propogate the localhost with a 'lazy port' redirect URI from the authorisation endpoint call to the token call.